dagfactory/dagbuilder.py

```diff
@@ -1117,8 +1124,12 @@ def adjust_general_task_params(task_params: dict(str, Any)):
         if utils.check_dict_key(task_params, "variables_as_arguments"):
             variables: List[Dict[str, str]] = task_params.get("variables_as_arguments")
             for variable in variables:
-                if Variable.get(variable["variable"], default_var=None) is not None:
-                    task_params[variable["attribute"]] = Variable.get(variable["variable"], default_var=None)
+                if INSTALLED_AIRFLOW_VERSION.major < AIRFLOW3_MAJOR_VERSION:
+                    if Variable.get(variable["variable"], default_var=None) is not None:
+                        task_params[variable["attribute"]] = Variable.get(variable["variable"], default_var=None)
+                else:
+                    if Variable.get(variable["variable"], default=None) is not None:
+                        task_params[variable["attribute"]] = Variable.get(variable["variable"], default=None)
```

Comment:
This version may lead to less repetition and highlight what is the Airlfow 3 breaking change:
```suggestion
                default_argument_name = "default"
                if INSTALLED_AIRFLOW_VERSION.major < AIRFLOW3_MAJOR_VERSION:
                    default_argument_name = "default_var"
                variable_value = Variable.get(variable["variable"], **{default_argument_name: None})
                if variable_value is not None:
                        task_params[variable["attribute"]] = variable_value
```